installer/install.go

```diff
@@ -0,0 +1,397 @@
+
+		HttpProxyClient struct {
+			Image string `yaml:"image,omitempty""`
+		} `yaml:"httpproxy-client"`
+
+		EchoClient struct {
+			Image string `yaml:"image,omitempty""`
+		} `yaml:"echo-client"`
+
+		Nats struct {
+			Image string `yaml:"image,omitempty""`
+		} `yaml:"nats"`
+
+		ImageRegistry struct {
+			Name string `yaml:"name,omitempty""`
+		} `yaml:"imageRegistry"`
+
+		Astra struct {
+			Token       string `yaml:"token"`
+			ClusterName string `yaml:"clusterName"`
+			AccountId   string `yaml:"accountId"`
+			AcceptEula  bool `yaml:"acceptEULA"`
+		} `yaml:"astra"`
+	} `yaml:"spec"`
+}
+
+func checkFatalErr(err error) {
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func getDockerUsername() (string, error) {
+	reader := bufio.NewReader(os.Stdin)
+	fmt.Print("Docker username: ")
+	username, err := reader.ReadString('\n')
+	if err != nil {
+		return "", err
+	}
+	return username, nil
+}
+
+func getDockerPw() (string, error) {
+	fmt.Print("Docker password: ")
+	bytePassword, err := term.ReadPassword(int(syscall.Stdin))
+	if err != nil {
+		return "", nil
+	}
+
+	return string(bytePassword), nil
+}
+
+func loadImageTar(dockerClient *client.Client, opts Options) ([]string, error) {
+	// We use system.OpenSequential to use sequential file access on Windows, avoiding
+	// depleting the standby list un-necessarily. On Linux, this equates to a regular os.Open.
+	file, err := system.OpenSequential(opts.ImageTar)
+	checkFatalErr(err)
+	defer file.Close()
+
+	imageLoadResponse, err := dockerClient.ImageLoad(context.Background(), file, true)
+	if err != nil {
+		log.Fatal(err)
+	}
+	if imageLoadResponse.Body == nil {
+		return nil, fmt.Errorf("docker load returned empty response")
+	}
+	if !imageLoadResponse.JSON {
+		return nil, fmt.Errorf("docker response is in unknown format")
+	}
+
+	var imageList []string
+	// Parse response, check for errors
+	scanner := bufio.NewScanner(imageLoadResponse.Body)
+	for scanner.Scan() {
+		imageResponseBody := DockerLoadResponse{}
+		log.Info(scanner.Text())
+		err := json.Unmarshal(scanner.Bytes(), &imageResponseBody)
+		checkFatalErr(err)
+
+		// Check response error
+		if imageResponseBody.Error != "" {
+			log.WithFields(log.Fields{
+				"error":       imageResponseBody.Error,
+				"errorDetail": imageResponseBody.ErrorDetail,
+			}).Fatal("Docker load error")
+		}
+
+		// Get loaded image names
+		imageName := strings.Split(strings.Trim(imageResponseBody.Stream, " \n"), ": ")[1]
+		imageList = append(imageList, imageName)
+		log.WithFields(log.Fields{
+			"image": imageName,
+		}).Info("Loaded image")
+	}
+	return imageList, nil
+}
+
+func tagImages(dockerClient *client.Client, images []string, repoPrefix string) ([]string, error) {
+	var imagesWithRepo []string
+	for _, image := range images {
+		//todo: update operator to leave the repo in place)E.g. 'myPrivateRepo/asdf.com/theotw/echo-proxylet:1.2.3'
+		newTag := fmt.Sprintf("%s/%s", repoPrefix, strings.TrimPrefix(image, "theotw/"))
```

Comment:
probably fine for now, but we may load from other places in the future

Reply:
Agreed. It will need a tweak in the operator